### PR TITLE
Updated for compatibility with django 3.2

### PR DESCRIPTION
**Pull request by frenetic-be**
#### Motivation for this PR

We need to upgrade Django in Picasso to version 3.2 LTS (https://app.asana.com/0/1200866257189754/1201981010595296/f). In this PR, we make sure our fork of the django-gdpr-assist package is compatible with Django 3.2.

#### Type of change

- [ ] Bug fix (fixes an issue)
- [ ] New feature (adds functionality)
- [ ] Refactor
- [x] Other: Library update

#### Related Asana task

https://app.asana.com/0/1200866257189754/1201981010595296/f

#### Virtual "desk check"

> If appropriate, consider including screenshots or a Loom video

#### Guidance for reviewers

- Is there a core set of ideas for which you are seeking feedback?

- What turnaround time are you looking for?
    - [ ] Not urgent
    - [x] In the coming days
    - [ ] Fairly urgent

- What should be reviewed by PM / designers before deploy?

#### Technical implications

- [ ] New technologies or new patterns were used in this PR
- [ ] This PR increases tech debt
- [ ] This PR decreases tech debt

##### Automated tests

- [ ] Modified code is covered by automated tests
- [ ] Modified code is partly covered by automated tests
- [ ] Modified code is not covered by automated tests at all

##### Other tests

- [ ] Modified code was tested locally
- [ ] A bug bash was conducted. If so, please link to a document with explicitly listed test cases
- For functionality uncovered by tests, how was it verified?

#### Deployment risk

How risky is this PR to deploy?

#### Roll out

- [ ] Feature flag
- [ ] 3-step migration
- [ ] All at once

#### Security and performance

- [ ] Changes in this PR might dramatically impact performance
- [x] Security impact of change has been considered. See [this page](https://owasp.org/www-project-top-ten/) for common web applications security risks. In particular, "Sensitive Data Exposure" and "Broken Access Control" are frequently relevant to application changes.
- [x] Code follows [company security practices and guidelines](https://docs.google.com/document/d/1XxDzlVGWBwIv7FD54tkuSPufhtDAM3NWwSto0rzUsco/edit?usp=sharing)

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -52,6 +52,7 @@ def runtests(args):
             }],
             GDPR_CAN_ANONYMISE_DATABASE=True,
             ROOT_URLCONF='tests.gdpr_assist_tests_app.urls',
+            SECRET_KEY='1234'
```
Comment (frenetic-be) [inline setup.py]: Running the tests was raising an [ImproperlyConfigured](https://docs.djangoproject.com/en/4.0/ref/exceptions/#django.core.exceptions.ImproperlyConfigured) exception due to https://docs.djangoproject.com/en/4.0/releases/3.2/#security.

File: gdpr_assist/models.py [hunk 2]
```diff
@@ -419,7 +418,7 @@ def get_anonymisation_tree(cls, prefix="", doprint=False, objs=[]):
             try:
                 field = getattr(cls, flat_field)
 
-                res += "%s|-> %s\n" % (prefix, field.field_name)
+                res += "%s|-> %s\n" % (prefix, flat_field)
```
Comment (frenetic-be) [inline gdpr_assist/models.py:421]: `field_name` is no longer an attribute of the DeferredAttribute object (https://docs.djangoproject.com/en/3.2/_modules/django/db/models/query_utils/)

File: gdpr_assist/admin/base.py [hunk 3]
```diff
@@ -10,6 +10,7 @@
 
 from django.conf.urls import url
 from django.contrib import admin
+from django.contrib.admin.helpers import ACTION_CHECKBOX_NAME
```
Comment (frenetic-be) [inline gdpr_assist/admin/base.py:13]: Compatibility import of `ACTION_CHECKBOX_NAME` was removed: https://docs.djangoproject.com/en/3.2/_modules/django/db/models/query_utils/

**Comment by jasonm** [comment]
@frenetic-be You mentioned running the tests -- since we don't have CI configured for this repo, I assume you did this locally and they pass?

**Review by jasonm — APPROVED** [review]
Assuming the tests pass, LGTM!

**Comment by jn123456789** [comment]
@jasonm wrote:
> @frenetic-be You mentioned running the tests -- since we don't have CI configured for this repo, I assume you did this locally and they pass?

Since this library is a dependency of picasso, and since we have unit tests in picasso that exercise this library code, I'd say therefore the changes here are pretty well covered.

**Review by jn123456789 — APPROVED** [review]
LGTM

**Comment by frenetic-be** [comment]
@jasonm Yes, I did run the tests locally with `python setup.py test`. In the latest commits, I edit the tests and the test settings to silence the warnings that we get when running the tests and I added a note in the README about running the tests.

**Pull request merged**